kalite/coachreports/views.py

```diff
@@ -155,7 +155,7 @@ def tabular_view(request, report_type="exercise"):
     student_ordering = ["last_name", "first_name", "username"]
 
     # Get a list of topics (sorted) and groups
-    topics = [get_node_cache("Topic", language=language).get(tid["id"]) for tid in get_knowledgemap_topics(language=language) if report_type.title() in tid["contains"]]
+    topics = [topic for (topicname, topic) in get_node_cache("Topic", language=language).iteritems() if (report_type.title() in topic.get("contains", [])) and (topic.get("render_type") == "Topic")]
```

Comment:
This list comprehension is getting a bit complex. Can you add in a comment on what we're searching for in here? The comment above doesn't really say much.